odoo/tools/sql.py

```diff
@@ -147,6 +171,9 @@ def __bool__(self):
     def __eq__(self, other):
         return isinstance(other, SQL) and self.code == other.code and self.params == other.params
 
+    def __hash__(self):
+        return hash(self.code) ^ sum(hash(p) for p in self.params)
```

Comment:
BTW hashing looks like a bad idea to me.
First, `self.params` may contain mutable data such as lists.
Second, you don't take `__to_flush` into account, which may lead to algorithms that "lose" fields to flush simply because some existing key doesn't have any.
Third, we were asked for this feature in the past, and we always found "saner" approaches (use a field or field name instead of `SQL.identifier`, etc).